[SDESK-7830] Upgrade Node to `v22` and drop support for `v14`

eos87 opened the pull request:
### Purpose
Migrate superdesk-client-core to Node `v22` and drop support for Node `v14`

### What has changed
  - **Node.js upgrade** - Bumped to Node v22 and dropped support for v14, updated all package-lock.json files to lockfileVersion 3
  - **ESLint upgrade** - Bumped ESLint and TypeScript ESLint to v8 with some new rules disabled for backward compatibility
  - **Mac compatibility fix** - Updated keyboard modifier tests so they work correctly on Mac
  - **Dependencies refresh** - Regenerated package-lock.json for the end-to-end testing helpers
  - **CI updates** - Upgraded GitHub Actions workflow to use the latest action versions (checkout v6, setup-node v6) 
  - Added package-lock to `e2e/client`

[SDESK-7830]

[SDESK-7830]: https://sofab.atlassian.net/browse/SDESK-7830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

eos87 committed:
Update Node to `v22` and drop support for `v14`

The tests workflow now runs only on Node.js v22 and uses 'npm ci' instead of 'npm install' for faster, more reliable installs. Updated package-lock.json files to lockfileVersion 3 for compatibility with newer npm versions.

SDESK-7830

eos87 committed:
Update keyBindingFn tests for Mac modifier keys

This fix the test so it runs on mac too

eos87 committed:
Upgrade ESLint and TypeScript ESLint dependencies

Updated ESLint and eslint-plugin-react to v8, and @typescript-eslint packages to v8 for compatibility with newer linting rules. Disabled several new ESLint rules in .eslintrc.js for backward compatibility and added TODOs for future evaluation.

SDESK-7830

eos87 committed:
Update `package-lock.json` in `end-to-end-testing-helpers`

SDESK-7830

eos87 committed:
Update tests.yml

eos87 committed:
Update GitHub Actions to use latest versions

thecalcc commented on scripts/extensions/helloWorld/yarn.lock:
```diff
@@ -2,104 +2,260 @@
 # yarn lockfile v1
 
 
+"@cspotcode/source-map-support@^0.8.0":
```
why the yarn lock?

thecalcc replied:
i mean, we could probably just drop it all together

eos87 replied:
it was already there. I will drop it later but also update documentation about creating extensions

thecalcc commented on package.json:
```diff
@@ -170,7 +167,10 @@
     "verify-client-api-changes": "node tasks/verify-client-api-changes.js"
   },
+  "engines": {
+    "node": ">=22.0.0"
```
i think if we specify the version in the GH workflow file we don't really need this, I'd personally leave it out, not wrong though

eos87 replied:
done

petrjasek commented:
it should be possible now to use package-lock also for e2e/client

eos87 committed:
Enable package-lock.json for e2e client

Removed .npmrc to allow generation of package-lock.json and added the lock file for dependency management in e2e/client.

SDESK-7830

eos87 committed:
Remove Node.js engine version requirement

SDESK-7830

eos87 commented:
Related with https://github.com/superdesk/superdesk-ui-framework/pull/997

petrjasek commented:
might be good to also setup workspaces in package.json so all those sub packages get auto installed and we see any possible conflicts right away

eos87 commented:
> might be good to also setup workspaces in package.json so all those sub packages get auto installed and we see any possible conflicts right away

I'll provide this in a separate PR.
